### PR TITLE
Fix time handling for dd_mon_yy format date strings

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix time handling for dd_mon_yy format

Before
----------------------------------------
`dd_mon_yy` format drops the time (if present) when processing the date

After
----------------------------------------
`dd_mon_yy` format keeps  the (if present)  time when processing the date

Technical Details
----------------------------------------
This PR also has the smaller fix in https://github.com/civicrm/civicrm-core/pull/28845

It also switches the month comparison from strtolower to mb_strtolower. However, that will have no impact until the validation regex is changed - which is in the last PR in this series https://github.com/civicrm/civicrm-core/pull/28716

Comments
----------------------------------------
_

### CODE REVIEW

File: CRM/Utils/Date.php [hunk 1]
```diff
@@ -2447,9 +2451,11 @@ protected static function getDateElement(string $date, string $separator, int $m
    * @param string $separator
    * @param int $yearPlacement
    *
+   * @internal
+   *
    * @return string
    */
-  public static function replaceShortYear($date, string $separator, int $yearPlacement): string {
+  protected static function replaceShortYear($date, string $separator, int $yearPlacement): string {
```
Comment (eileenmcnaughton) [inline CRM/Utils/Date.php:2458]: This function & the following function were only just added. We should make protected & internal unless there is a reason not to so fixing in this PR

**Comment by eileenmcnaughton** [comment]
CRM_Contact_Import_Parser_ContactTest::testValidateDateData with data set "type_8" ('individual_dates_type8.csv', 8)
September 12, 2008 should have been valid

/home/homer/buildkit/build/build-3/web/sites/all/modules/civicrm/tests/phpunit/CRM/Contact/Import/Parser/ContactTest.php:2343
/home/homer/buildkit/build/build-3/web/sites/all/modules/civicrm/tests/phpunit/CRM/Contact/Import/Parser/ContactTest.php:1472
/home/homer/buildkit/build/build-3/web/sites/all/modules/civicrm/tests/phpunit/CiviTest/CiviUnitTestCase.php:247
/home/homer/buildkit/extern/phpunit9/phpunit9.phar:2307

**Comment by colemanw** [comment]
Good amount of code comments and unit test coverage.

**Comment by eileenmcnaughton** [comment]
Thanks @colemanw  - there is one last PR in this series... https://github.com/civicrm/civicrm-core/pull/28716

**Pull request merged**